packages/api/endpoints/rules.js

```diff
@@ -60,26 +64,28 @@ async function get(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function post(req, res) {
+  const {
+    model = new models.Rule(),
+    dbClient = await getKnexClient(),
+  } = req.testContext || {};
+
+  let record;
+  const rule = req.body || {};
+  const name = rule.name;
+
+  if (await model.exists(name)) {
+    return res.boom.conflict(`A record already exists for ${name}`);
+  }
+
+  const postgresRecord = await translateApiRuleToPostgresRule(rule, dbClient);
```

Comment:
Sorry to be annoying - but let's move this inside the try but above the transaction. That way if this function did throw an error, we'd get a deterministic endpoint response based on the `catch` logic

Reply:
Not annoying at all! that's a good point, will move it.